### PR TITLE
Implemented new version for clustering

### LINKED ISSUE #198: implement new way of clustering
As becomes apparent from this history, the clustering step actually fails to group corresponding peaks together, even if assigned larger tolerances: https://umsa.cerit-sc.cz/u/hechth/h/20230412-rcx-aplcms-qc-new-parameters (see also the screenshot below).

The peaks simply get assigned to different clusters and therefore the number of peaks in the aligned peak table is significantly smaller than in the individual peak tables, where the peaks that are detected are quite comparable for comparable files. The clustering algorithm has to be revisited or a new one has to be implemented to determine which features are shared across samples.

image

### LINKED ISSUE #199: update test data for extract-features full data test case
The expected output still contains the `pos` column instead of rt - this should be renamed to `rt`.

### PR DESCRIPTION
**Pull request by hechth**
- Implemented new simple clustering computation method based on only numerical tolerances and using dplyr operations.
- Implemented unit tests for new clustering method
- Implemented unit tests for find_mz_tolerance
- Implemented unit test for find.tol.time

Closes #198 
Closes #199

### CODE REVIEW

**Review by xtrojak — COMMENTED** [review]
@hechth Why did you remove all the `.Rd` files in https://github.com/RECETOX/recetox-aplcms/pull/201/commits/cec9f8ae0ebd169f624d72c38e6b65f6f9746489?

**Comment by hechth** [comment]
> @hechth Why did you remove all the `.Rd` files in [cec9f8a](https://github.com/RECETOX/recetox-aplcms/commit/cec9f8ae0ebd169f624d72c38e6b65f6f9746489)?

They are auto generated anyway, so I think they just increase the repository size unnecessarily

**Comment by xtrojak** [comment]
> > @hechth Why did you remove all the `.Rd` files in [cec9f8a](https://github.com/RECETOX/recetox-aplcms/commit/cec9f8ae0ebd169f624d72c38e6b65f6f9746489)?
> 
> They are auto generated anyway, so I think they just increase the repository size unnecessarily

I thought they are maybe needed somewhere, but I guess you are right. Would be nice to note this in the changelog.

File: R/compute_clusters.R [hunk 1]
```diff
@@ -74,37 +74,104 @@ compute_clusters <- function(feature_tables,
...
 }
 
+#' Compute clusters using simple grouping based on numeric thresholds.
+#' 
+#' @describtion
+#' Features are first grouped in mz dimension based on the tolerance.
+#' First, the absolute tolerance is computed for each feature, then a new group is started
+#' once the difference between consecutive features is above this threshold.
+#' The same process is then repeated for the retention time dimension.
+#' The individual indices and then combines into a single index in the `cluster` columns.
```
Comment (xtrojak) [inline R/compute_clusters.R]:
```suggestion
#' The individual indices are then combined into a single index in the `cluster` columns.
```

File: tests/testthat/test-find.tol.time.R [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    breaks,
+    max.num.segments,
+    aver.bin.size,
+    number_of_samples,
+    rt,
+    min.bins,
+    max.bins
+  )
+
+  expect_equal(actual, 1.04166666666667)
```
Comment (xtrojak) [inline tests/testthat/test-find.tol.time.R]: It may be better to specify that the float number is within a range.

**Review by xtrojak — APPROVED** [review]
Good job with the rewrite of clustering, looks awesome!

File: tests/testthat/test-find.tol.time.R (hunk 2 above)
Comment (hechth) [reply tests/testthat/test-find.tol.time.R]:
```suggestion
  expect_equal(actual, 1.0416)
```

File: R/compute_clusters.R [hunk 3]
```diff
@@ -74,37 +74,104 @@ compute_clusters <- function(feature_tables,
...
 
+#' Compute clusters using simple grouping based on numeric thresholds.
+#' 
+#' @describtion
+#' Features are first grouped in mz dimension based on the tolerance.
+#' First, the absolute tolerance is computed for each feature, then a new group is started
+#' once the difference between consecutive features is above this threshold.
+#' The same process is then repeated for the retention time dimension.
+#' The individual indices are then combined into a single index in the `cluster` columns.
+#' @param feature_tables list of tibbles List of feature tables coming from all samples.
```
Comment (zargham-ahmad) [inline R/compute_clusters.R]:
```suggestion
#' @param feature_tables list of tibbles feature tables coming from all samples.
```

File: tests/testthat/test-find.tol.time.R [hunk 4]
```diff
@@ -0,0 +1,21 @@
+test_that("compute_rt_tol_relative computes something", {
```
Comment (zargham-ahmad) [inline tests/testthat/test-find.tol.time.R:1]: "computes something"?

**Review by zargham-ahmad — APPROVED** [review]
Looks good!

**Comment by zargham-ahmad** [comment]
@hechth Does deleting man files really make any difference in the repo size because they were around 80 kb anyways? Also, I think the CI seems to be failing because of it `No man pages found in package  ‘recetox.aplcms’ `

**Pull request merged**